nilearn/regions/rena_clustering.py

```diff
@@ -703,30 +705,50 @@ def fit(self, X, y=None):
         """
         del y
         check_params(self.__dict__)
-        X = check_array(
-            X, ensure_min_features=2, ensure_min_samples=2, estimator=self
-        )
-        n_features = X.shape[1]
 
+        if not SKLEARN_LT_1_6:
+            from sklearn.utils.validation import validate_data
+
+            X = validate_data(
+                self,
+                X,
+                reset=True,
+                ensure_min_features=2,
+                ensure_min_samples=2,
+            )
+        else:
+            X = check_array(
+                X, ensure_min_features=2, ensure_min_samples=2, estimator=self
+            )
+            self.n_features_in_ = X.shape[1]
+
+        # If no mask images was passed we create a dummy nifti image
+        # with a single slice
+        self.mask_img_ = self.mask_img
+        if self.mask_img_ is None:
+            self.mask_img_ = Nifti1Image(
+                np.ones((self.n_features_in_, 1, 1), dtype=np.int8),
+                affine=np.eye(4),
+            )
```

Comment:
@bthirion 
I am adding this to make it easier to ensure the compliance of the estimator with sklearn rules.

However this means that the fit will run even if user do not provide a meaningful mask that was used to extract the data.
So it means the clustering of the data will be unrelated to the original 'topology' of the data.

I am not really happy with this.

At minimum, I think we should throw a warning.

Otherwise I can try to hack something, so this dummy image is only created during tests.

what do you think?

Reply:
added a _set_mask_img_for_tests method

it's ugly because we have a slightly different behavior in tests but it does make it easier to test sklearn compliance.